openapi.go

```diff
@@ -47,35 +58,79 @@ func (s *Server) GenerateOpenAPI() {
 	}
 
 	// Write spec to docs/openapi.json
-	err = os.MkdirAll("docs", 0o750)
-	if err != nil {
-		slog.Error("Error creating docs directory", "error", err)
-	}
-	f, err := os.Create("docs/openapi.json")
-	if err != nil {
-		slog.Error("Error creating docs/openapi.json", "error", err)
+	if options.disableSaveOnDisk == false {
+		jsonLocalPath := "docs/openapi.json"
+		if options.jsonLocalPath != "" {
+			jsonLocalPath = options.jsonLocalPath
+
+			match, _ := regexp.MatchString(`^[\.a-zA-Z0-9\-\_]+(.json)$`, jsonLocalPath)
+
+			if match == false {
+				err := errors.New("Value of 'jsonLocalPath' option is not valid")
+				slog.Error("Error writing json spec", "error", err)
```

Comment:
It's a bit confusing to create an error when it's just being used in a log, so I think it's better to write it directly.